src/porepy/numerics/fv/upwind.py

```diff
@@ -109,7 +109,7 @@ def assemble_matrix_rhs(
                 with multiple components is only supported in Ad mode.
                 """
             )
-        rhs = div * (bc_discr_neu + bc_discr_dir * flux_mat) * bc_values
+        rhs = -div * (bc_discr_neu + bc_discr_dir * flux_mat) * bc_values
```

Comment:
This I am not sure if required, and how it is related to the modifications in the tests for upwinding.
Please check.

Reply:
This change has been undone and the three failing tests have been updated.